### DISCUSSION TITLE
Question on RNEA()

### DISCUSSION THREAD

**Discussion by xiaosuyang** [stack-of-tasks/pinocchio#2225] [Q&A] [answered]
Hello.
How to use the function:
pinocchio::rnea(model,data,q,v,a,fext) to calculate the joint torque. I want to add the external spatial force to a "fixed joint" in the urdf file of a biped robot.  How to set the para const container::aligned_vector<ForceDerived>& fext

**Comment by MegMll** [comment, marked as answer]
Hello, 

In pinocchio, "fixed joint" are not considered as joints, but only as frames. So it's not possible to directly add the external force on this joint, because it does not exist in pinocchio's model.

What you could do, is add this force on the parentJoint of the "fixed joint", in the fext vector. 

For Rnea, fext is a vector of external [forces](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/classpinocchio_1_1ForceTpl.html) expressed in the local frame of the joints. 

So if you know, the external force that is applied on the "fixed_joint", you just need to get its expression into the parentJoint.

**Reply by xiaosuyang** [reply]
Thanks for reply. I want to use the rnea(nodel,data,q,v,a,fext) on a 12 dof biped robot. But I find that the tau computed by the rnea() include the freeflyerjoint, in other word the tau of freeflyerjoint is not zero, so the taus of the bip robot's joints  are not correct, how can I do to let the torque of freeflyerjoint to be zero when using rnea()

**Reply by xiaosuyang** [reply]
I seem to understand. The rnea() would include the freeflyerjoint ‘s torque for blance and I should compute the right fext before calling rnea()

**Reply by jcarpent** [reply]
RNEA does not include the fact that the free-floating is not actuated. You should rather solve an equality-constrained inverse-dynamics problem, which can be done leveraging pinocchio of course.
